test/sqllogictest/table_func.slt

```diff
@@ -242,7 +242,6 @@ EXPLAIN PLAN FOR SELECT * FROM x x1, x x2, generate_series(x1.a, x2.a) WHERE x1.
 %2 =
 | Join %0 %1
 | | implementation = Differential %1 %0.()
-| | demand = (#0..#3)
```

Comment:
I'm a bit confused about these, sorry. There seem to be no projection changes, and yet the demand goes away. That seems to be a loss of information. I know you said elsewhere that it shouldn't be needed, but this looks like something "suboptimal" happens here. Not the end of the world probably, but not expected, right? Is there an optimization here that `Demand` sees that projection pushdown does not see?

Reply:
`(#0..#3)` is all the columns from all inputs of the join. Thus, adding a projection around the join is unnecessary.

Reply:
Ah, I'm confused about the change for a different reason then. I thought you mentioned elsewhere that you didn't change the `demand` printing, and it just happens that when it is all the columns it is suppressed. This example doesn't seem to have a changed number of columns (no other changes than this line) so I'm less clear about what happened. Did it go from "demand all the columns explicitly" to "I have no demand information"? If so, were we just lucky that producing no demand information worked out on this join, and it could have been a problem?

There are a few other "projection after join" and "demand" questions I had, so happy to wait for them to be shaken out before trying to figure out which of them if any reflect actual problems.